distributed/deploy/tests/test_local.py

```diff
@@ -1050,28 +1050,31 @@ async def test_cluster_names():
             assert unnamed_cluster2 != unnamed_cluster
 
 
-@pytest.mark.asyncio
 @pytest.mark.parametrize("nanny", [True, False])
+@gen_test()
 async def test_local_cluster_redundant_kwarg(nanny):
-    with pytest.raises(TypeError, match="unexpected keyword argument"):
-        # Extra arguments are forwarded to the worker class. Depending on
-        # whether we use the nanny or not, the error treatment is quite
-        # different and we should assert that an exception is raised
-        async with await LocalCluster(
-            typo_kwarg="foo",
-            processes=nanny,
-            n_workers=1,
-            dashboard_address=":0",
-        ) as cluster:
+    cluster = LocalCluster(
+        typo_kwarg="foo",
+        processes=nanny,
+        n_workers=1,
+        dashboard_address=":0",
+        asynchronous=True,
+    )
+    try:
+        with pytest.raises(TypeError, match="unexpected keyword argument") as exc_info:
+            # Extra arguments are forwarded to the worker class. Depending on
+            # whether we use the nanny or not, the error treatment is quite
+            # different and we should assert that an exception is raised
+            async with cluster:
+                pass
+    finally:
+        # FIXME: LocalCluster leaks if LocalCluster.__aenter__ raises
+        await cluster.close()
 
-            # This will never work but is a reliable way to block without hard
```

Comment:
I guess this is removed because the `async with cluster` in the modified version reliably blocks for this test now?

Reply:
the code didn't need any blocking at all - it was unreachable because cluster raises the TypeError in the `__aenter__`

Reply:
the code was added here, was it reachable when that PR was made and then later made unreachable? 

https://github.com/dask/distributed/pull/4733/files#diff-c9c4da87696cc1b04f54c8b199ebd277ca69da7eefcc08109d64349a4d2309a9R1081 

cc @fjetter

Reply:
The client in this test was never reachable. It was never intended to be reachable. I can't remember why I wrote the test in such a complicated fashion. I'm OK with the new version